BuildScripts/mvBESFunctions.sql

```diff
@@ -0,0 +1,240 @@
+Routine Name: mv$renameMaterializedView
+Author:       Rohan Port
+Date:         17/08/2021
+------------------------------------------------------------------------------------------------------------------------------------
+Revision History    Push Down List
+------------------------------------------------------------------------------------------------------------------------------------
+Date        | Name          | Description
+------------+---------------+-------------------------------------------------------------------------------------------------------
+            |               |
+------------+---------------+-------------------------------------------------------------------------------------------------------
+Description:    Renames a materialized view, edits the view_name in the pg$mviews table
+
+                This function performs the following steps
+                1)  Edits the view_name in the pg$mviews table to be the new name
+                2)  Alters the materialized view table name to be the new name
+
+Arguments:      IN      pOldViewName        The existing name of the materialized view
+                IN      pNewViewName        The new name of the materialized view
+                IN      pOwner              Optional, the owner of the materialized view, defaults to user
+Returns:                VOID
+************************************************************************************************************************************
+Copyright 2021 Beyond Essential Systems Pty Ltd
+***********************************************************************************************************************************/
+DECLARE
+
+    aPgMview    pg$mviews;
+    rConst      mv$allConstants;
+
+    tUpdatePgMviewsSqlStatement   TEXT := '';
+    tRenameTableSqlStatement      TEXT := '';
+    tRenameIndexSqlStatement      TEXT := '';
+   
+    rIndex	         RECORD;
+    tOldIndexName    TEXT;
+    tNewIndexName    TEXT;
+begin
+	
+	rConst      := mv$buildAllConstants();
+
+    tUpdatePgMviewsSqlStatement   :=  rConst.UPDATE_COMMAND || 'pg$mviews' || rConst.SET_COMMAND || 'view_name = ' 
+                                           || rConst.SINGLE_QUOTE_CHARACTER || pNewViewName || rConst.SINGLE_QUOTE_CHARACTER 
+                                           || rConst.WHERE_COMMAND || 'view_name = ' || rConst.SINGLE_QUOTE_CHARACTER || pOldViewName || rConst.SINGLE_QUOTE_CHARACTER;
+    
+    tRenameTableSqlStatement   :=  rConst.ALTER_TABLE || pOldViewName || rConst.RENAME_TO_COMMAND || pNewViewName;
+
+    EXECUTE tUpdatePgMviewsSqlStatement;
+    EXECUTE tRenameTableSqlStatement;
+
+    FOR rIndex IN 
+        SELECT indexname FROM pg_indexes WHERE schemaname = pOwner AND tablename = pNewViewName AND indexname like '%' || rConst.MV_M_ROW$_COLUMN || '%'
+    LOOP
+        tOldIndexName := rIndex.indexname;
+        tNewIndexName := REPLACE(tOldIndexName, pOldViewName, pNewViewName);
```

Comment:
Technically this isn't the safest operation... Let's say you had an old table with name `a` then this would replace all `a`'s in the index name with the new table name... We could make this stricter, but I was getting a tad lazy and felt it might be overkill...

Reply:
Thanks for pointing this out, I definitely wouldn't have spotted it. 

Just thinking out loud...other than a small table name that might show up in subwords, another potential risk is if there was some column that we were indexing on that included the table name. It's not common practice for obvious reasons, but does happen sometimes (closest I can think of in our db is `sync_service.service_type`.

So you might get (stretching to an unlikely case) `analytics.analytics_type` or `analytics.analytics_format`, and then an index on that would have a name like `analytics_analytics_format` which would become e.g. `temp_temp_format`. 

In the end though, the index name doesn't even matter too much, so I say leave it as is!

Reply:
You do make a good point... I think given our current table and our current use case I can never see it actually becoming and issue. That being said I don't think I have much of an excuse outside of laziness here 😜 If I knew more about string manipulation in postgres I would have done the right thing